src/main/java/ch/uzh/ifi/hase/soprafs21/objects/GameEngine.java

```diff
@@ -331,26 +336,30 @@ public void createGameFromGameSessionAndFillUp(GameSession gameSession){
         int numberOfUsersToInvite = PLAYER_AMOUNT - gameSession.getUserList().size();
 
         if(!gameSession.getInvitedUsers().isEmpty()){
-            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Cant use FillUp while there are still pending gameRequests");
-        }
-        else if(numberOfUsersToInvite > waitingRoom.getUserCount()){
-            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "There are too few users in WaitingRoom to fillUp");
+            String msg =  "Cant use FillUp while there are still pending gameRequests";
+            webSocketService.sendGameSessionFillUpError(userService.convertUserNameToSessionIdentity(gameSession.getHostName()), msg);
         }
         else {
 
-            List<User> usersFromWaitingRoomToFillup = waitingRoom.getXNumberOfUsers(numberOfUsersToInvite);
-            List<User> usersInGameSession = gameSession.getUserList();
+            try {
+                List<User> usersFromWaitingRoomToFillup = waitingRoom.getXNumberOfUsers(numberOfUsersToInvite);
+                List<User> usersInGameSession = gameSession.getUserList();
 
-            gameSession.getUserList().addAll(usersFromWaitingRoomToFillup);
-            Game createdGame = createGameFromGameSession(gameSession);
+                gameSession.getUserList().addAll(usersFromWaitingRoomToFillup);
+                Game createdGame = createGameFromGameSession(gameSession);
 
-            for (User userForFillup : usersFromWaitingRoomToFillup) {
-                String userIdentity = userForFillup.getSessionIdentity();
-                webSocketService.sendGameAssignmentMessageToWaitingRoom(userIdentity, createdGame.getPlayers(), createdGame.getGameId());
+                for (User userForFillup : usersFromWaitingRoomToFillup) {
+                    String userIdentity = userForFillup.getSessionIdentity();
+                    webSocketService.sendGameAssignmentMessageToWaitingRoom(userIdentity, createdGame.getPlayers(), createdGame.getGameId());
+                }
+                for(User usersGameSession: usersInGameSession){
+                    String userIdentity = usersGameSession.getSessionIdentity();
+                    webSocketService.sendGameAssignmentMessageToGameSession(userIdentity, createdGame.getPlayers(), createdGame.getGameId());
+                }
             }
-            for(User usersGameSession: usersInGameSession){
-                String userIdentity = usersGameSession.getSessionIdentity();
-                webSocketService.sendGameAssignmentMessageToGameSession(userIdentity, createdGame.getPlayers(), createdGame.getGameId());
+            catch (Exception e){
+                String msg = "There are too few users in WaitingRoom to fillUp";
```

Comment:
mmmh würd diese message direkt auf die Exception packen, wenn du sie throws in getXNumberOfUsers(), mit throw new Exception("There are too few users in WaitingRoom to fillUp");, und dann kannst du sie hier mit e.getMessage() der webSocketService.sendGameSessionFillUpError weitergeben. So bleibt der code besser erweiterbar, wenn im try block mal eine andere stelle eine exception wirft, bleibt die message dynamisch

Reply:
hast du völlig recht, besser so, done